File: windows/Scanner/Rules/AESCFG003.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESCFG003 : IRule
{
    public string Id => "AES-CFG-003";
    public Severity Severity => Severity.Low;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts) =>
        facts.McpServers
            .Where(server => !server.Disabled)
            .Where(server => string.IsNullOrWhiteSpace(server.Description))
            .Select(server => this.Finding(server.AppId, server.Name, affectedPath: server.ConfigPath))
            .ToArray();
}

File: macos/Scanner/Sources/Scanner/Rules/AESCFG003.swift
public struct AESCFG003: Rule {
    public let id = "AES-CFG-003"
    public let severity: Severity = .low

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        facts.mcpServers.compactMap { server in
            guard !server.disabled else {
                return nil
            }
            let description = server.description?.trimmingCharacters(in: .whitespacesAndNewlines)
            guard description?.isEmpty != false else {
                return nil
            }
            return finding(app: server.appId, serverName: server.name, affectedPath: server.configPath)
        }
    }
}
